### PR TITLE
Add partitioninfo module

### PR DESCRIPTION
**Pull request by srivathsa729**
Prints partition information

Authored-by: Gulam Mohamed <gulam.mohamed@oracle.com>
Co-authored-by: Srivathsa Dara <srivathsa.d.dara@oracle.com>

### CODE REVIEW

**Comment by srivathsa729** [comment]
Sample Output:
```
MAJOR  MINOR  NAME  START    SECTORS    READ-ONLY
251    0      vda   0        734003200  No
251    1      vda1  2048     2097152    No
251    2      vda2  2099200  312473567  No
11     0      sr0   0        2097151    No
252    0      dm-0  0        301981696  No
252    1      dm-1  0        10485760   No

File: drgn_tools/partitioninfo.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+from drgn.helpers.linux.block import part_name
+from drgn.helpers.linux.device import MAJOR
+from drgn.helpers.linux.device import MINOR
+from drgn.helpers.linux.xarray import xa_for_each
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.table import print_table
+
+
+def is_blkdev(prog: Program) -> bool:
```
Comment (biger410) [inline drgn_tools/partitioninfo.py:25]: Can you rename as "part_is_blkdev"?

File: drgn_tools/partitioninfo.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    """
+    Checks whether partition is represented by `struct block_device`
+    or `struct hd_struct`
+    """
+    for disk in for_each_partition(prog):
+        if str(disk.type_) == "struct hd_struct *":
+            return False
```
Comment (biger410) [inline drgn_tools/partitioninfo.py:32]: you can do something like this, no need check a real partition for it.
```
if "hd_struct" in disk.part0.type_.type_name()
```

File: drgn_tools/partition.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    Returns a list with partition information for the given partition.
+    """
+    devt = part.bd_dev.value_()
+    name = escape_ascii_string(part_name(part), escape_backslash=True)
+    start_sect = int(part.bd_start_sect)
+    nr_sects = int(part.bd_inode.i_size.value_() / 512)
+    if part.bd_read_only:
+        ro = "Yes"
+    else:
+        ro = "No"
```
Comment (biger410) [inline drgn_tools/partition.py:33]: python style, 
```
ro = "Yes" if part.bd_read_only else "No"

File: drgn_tools/partition.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    Returns a list with partition information for the given partition.
+    """
+    devt = part_devt(part)
+    name = part_name(part).decode()
+    start_sect = int(part.start_sect)
+    nr_sects = part.nr_sects.value_()
+    if int(part.policy):
+        ro = "Yes"
+    else:
+        ro = "No"
```
Comment (biger410) [inline drgn_tools/partition.py:57]: python style, 
```
ro = "Yes" if int(part.policy) else "No"
```

File: drgn_tools/partition.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    ]
+
+
+def get_partinfo_from_hd_struct(part: Object) -> list:
+    """
+    Collects partition information from `struct hd_struct`
+    Returns a list with partition information for the given partition.
+    """
+    devt = part_devt(part)
+    name = part_name(part).decode()
```
Comment (biger410) [inline drgn_tools/partition.py:51]: Wonderding why using "escape_ascii_string(part_name(part), escape_backslash=True)" in "get_partinfo_from_blkdev_struct" but using part_name(part).decode() here?

Comment (srivathsa729) [reply drgn_tools/partition.py:47]: No specific reason junxiao, i'm making both of use decode to make things consistent

**Review by biger410 — APPROVED** [review]
Looks good

**Pull request merged**